mitodl/ocw-studio
# separate online and offline parts of site-pipeline into separate jobs

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-studio/issues/1764

#### What's this PR do?
This PR takes the online and offline parts of `site-pipeline.yaml` and separates them into two separate jobs. A resource of type `keyval` was added, which is used as a simple gate between the jobs. At the end of the online build, the `timestamp` value in `offline-build-gate` is updated, which triggers the offline build.

There is also a change to how the mp4 filtering works. In the online build, mp4 files are filtered out the same way they always have been by removing them before the s3 sync takes place. In the offline build, the mp4 files are synced to the mirror drive buckets, but are excluded from the ZIP download that is uploaded to the live site.

#### How should this be manually tested?

 - Spin up `ocw-studio` on this branch, making sure you have Concourse, Minio, Google Drive and Github configured properly for publishing websites, as well as a recent database restore from RC
 - Create a folder somewhere on your computer and pull down the static resources for the website we'll be using from S3 with `aws s3 sync s3://ol-ocw-studio-app-qa/courses/6-041sc-probabilistic-systems-analysis-and-applied-probability-fall-2013/ ./` in a bash terminal in the folder you want to download them in, using the credentials from `ocw-studio` RC
 - Browse to http://localhost:9001 and log in to the Minio web UI
 - In your `ol-ocw-studio-app` bucket, create a folder structure matching what we synced above, creating a `courses` folder with a `6-041sc-probabilistic-systems-analysis-and-applied-probability-fall-2013` folder under that
 - Inside this folder, upload the static resources we downloaded above
 - Refresh the pipeline definition for our test course by running `docker compose exec web ./manage.py backpopulate_pipelines --filter 6-041sc-probabilistic-systems-analysis-and-applied-probability-fall-2013`
 - In the `ocw-studio` UI at http://localhost:8043/sites/6-041sc-probabilistic-systems-analysis-and-applied-probability-fall-2013 publish our test site live
 - Browse to http://localhost:8080/teams/main/pipelines/live?vars.site=%226-041sc-probabilistic-systems-analysis-and-applied-probability-fall-2013%22 and confirm that the `build-online-ocw-site` job is running and that the `build-offline-ocw-site` job has not yet started
 - After the online job finishes, ensure that:
   - At http://localhost:8043/sites/6-041sc-probabilistic-systems-analysis-and-applied-probability-fall-2013/ you see a green light icon and the status is "succeeded"
   - The course site is available at http://localhost:8045/courses/6-041sc-probabilistic-systems-analysis-and-applied-probability-fall-2013/
 - After the offline job finishes, go back to the Minio UI at http://localhost:9001/ and:
   - Browse to `ocw-content-offline-live/courses/6-041sc-probabilistic-systems-analysis-and-applied-probability-fall-2013/static_resources` and verify that the folder contains the course videos
   - Browse to `ocw-content-live/courses/6-041sc-probabilistic-systems-analysis-and-applied-probability-fall-2013` and download the course .zip file. Extract the file and ensure that it does not contain .mp4 videos in the `static_resources` folder

#### Where should the reviewer start?
`content_sync/pipelines/definitions/site-pipeline.yml`

#### Any background context you want to provide?
This is being done as part of the work for https://github.com/mitodl/ocw-hugo-themes/issues/1133. It is desired that on mirror drive sites, the local archive video of any video resource is used instead of YouTube if it exists. Adding the archive videos to the offline build increases the build time significantly. This work is being done to avoid delays in publishing for users.

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/12089658/235560732-917c76ae-69c3-4f85-b2dc-f6d0a16d7e79.png)
